packages/rocketchat-language-plugin/server/server.js

```diff
@@ -0,0 +1,82 @@
+const parser = Npm.require('accept-language-parser');
+const languages = Npm.require('languages');
+
+function get_language(languages_codes) {
+
+	const languages_parsed = parser.parse(languages_codes);
+	if (languages_parsed.length===0) {
+		return;
+	}
+	const priority_language = languages_parsed[0].code;
+
+	const priority_language_fullname = languages.getLanguageInfo(priority_language);
+	return priority_language_fullname;
+}
+
+function remove_user_from_automatic_channel() {
+	const user_info = RocketChat.models.Users.findOneById(Meteor.userId());
+	// get the name of channels to which user is added
+	const to_be_removed = user_info.automatic_channels.filter(function(obj) {
+		return obj.blacklisted ==='false';
+	});
+	if (to_be_removed.length!==0) {
+		Meteor.users.update({
+			_id: Meteor.userId()
+		}, {
+			$pull: { automatic_channels: { name: to_be_removed[0].name } }
+		});
+		const room = RocketChat.models.Rooms.findOneByIdOrName(to_be_removed[0].name);
+		RocketChat.removeUserFromRoom(room._id, Meteor.user());
+		// delete the room if its the last user
+		if (room.usernames.length===1) {
+			RocketChat.models.Messages.removeByRoomId(room._id);
+			RocketChat.models.Subscriptions.removeByRoomId(room._id);
+			RocketChat.models.Rooms.removeById(room._id);
+		}
+
+	} else {
+		return;
+	}
+}
+Accounts.onLogin(function(user) {
+
+	if (!Meteor.userId()) {
+		return;
+	}
+
+	const browser_language = get_language(user.connection.httpHeaders['accept-language']);
```

Comment:
What if `browser_language` is not set because you returned from the function early?